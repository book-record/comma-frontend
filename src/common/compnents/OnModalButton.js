import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

function OnModalButton({ onClick, text }) {
  return (
    <ModalButton type="button" onClick={onClick}>
      {text}
    </ModalButton>
  );
}

const ModalButton = styled.button`
  font-family: "Nanum Gothic", sans-serif;
  font-size: 20px;
  font-weight: 700;
  margin: 10px 20px;
  border: none;
  background: none;
`;

export default OnModalButton;

OnModalButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
};
