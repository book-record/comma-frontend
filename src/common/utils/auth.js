const isLogined = () => !!localStorage.getItem("Authorization");
export default isLogined;
