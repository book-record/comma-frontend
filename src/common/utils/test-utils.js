/* eslint-disable react/prop-types */
import { configureStore } from "@reduxjs/toolkit";
import { render as rtlRender } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";

import recordRouter from "../../store/recordSlice";
import userRouter from "../../store/userSlice";

function render(
  ui,
  {
    preloadedState,
    store = configureStore({
      reducer: {
        user: userRouter,
        record: recordRouter,
      },
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";
export { render };
