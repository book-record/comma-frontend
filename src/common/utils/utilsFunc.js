exports.validateDate = (start, finish) => {
  if (finish.diff(start, "d") > 1) {
    return `D-day ${finish.diff(start, "d")}일`;
  }

  if (finish.diff(start, "h") > 0) {
    return `D-day ${finish.diff(start, "h")}시간`;
  }

  if (finish.diff(start, "s") > 0) {
    return `D-day ${finish.diff(start, "m")}분`;
  }

  return "D-day";
};
