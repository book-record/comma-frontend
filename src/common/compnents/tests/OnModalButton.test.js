import React from "react";

import { fireEvent, cleanup, render, screen } from "../../utils/test-utils";
import OnModalButton from "../OnModalButton";

afterEach(cleanup);

describe("<OnModalButton>", () => {
  const onClick = jest.fn();

  const { getByRole, container } = render(
    <OnModalButton onClick={onClick} text="보내다" />
  );

  it("OnModalButton props로 넘어온 값이 잘 넘어왔는지 확인", () => {
    expect(onClick).toHaveBeenCalledTimes(0);
    expect(screen.getByText("보내다"));
    expect(container).toMatchSnapshot();
  });

  it("OnModalButton 클릭시 클릭 여부 반영", () => {
    render(<OnModalButton onClick={onClick} text="보내다" />);
    const button = getByRole("button");

    fireEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
