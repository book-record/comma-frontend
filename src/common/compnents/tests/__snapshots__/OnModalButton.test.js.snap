// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OnModalButton> OnModalButton props로 넘어온 값이 잘 넘어왔는지 확인 1`] = `
<div>
  <button
    class="sc-bdvvtL fqMHbG"
    type="button"
  >
    보내다
  </button>
</div>
`;
